app/models/saved_claim/disability_compensation.rb

```diff
@@ -33,7 +34,7 @@ def to_submission_data(user)
                                                                                      @form_hash.deep_dup).translate,
       Form526Submission::FORM_8940 => EVSS::DisabilityCompensationForm::Form8940.new(user,
                                                                                      @form_hash.deep_dup).translate,
-      'flashes' => BGS::DisabilityCompensationFormFlashes.new(user, @form_hash.deep_dup).translate
+      'flashes' => BGS::DisabilityCompensationFormFlashes.new(user, @form_hash.deep_dup, claimed_disabilities).translate
```

Comment:
I don't love adding a new argument to this constructor, and I think it's mostly because the `claimed_disabilities` data is already available in the broader `form526` hash, albeit in a slight different format maybe.

Do you want to try extracting the disabilities from the form hash within `DisabilityCompensationFormFlashes`? You'd need to move some stuff around in the spec -- you could either cook up a new VCR cassette with ALS data, or have the spec contexts insert ALS data into the existing VCR cassette.

Reply:
Although you're right the `@form_hash` already has the disabilities in a different state, `EVSS::DisabilityCompensationFormDataTranslationAllClaim` already takes care of the extracting the selected disabilities of the claim, compiling a list of new and CFI into a single `disabilities` object. Adding this line to the constructor, prevents some degree of WET code for translating the disabilities from `form526`. 

The alternative for pulling out the `claimed_disabilities` from the `form_hash` would be:
* to construct another `EVSS::DisabilityCompensationForm::DataTranslationAllClaim` and call `translate_disabilities` (currently a private method) inside of `BGS::DisabilityCompensationFormFlashes` to get the disabilities, OR
* rewrite some the code contained within that `translate_disabilities` to convert the form hash to the list of claimed disabilities.

It seemed like digging the disabilities from the form in `disability_compensation.rb` was the most efficient way to get them into the `BGS::DisabilityCompensationFormFlashes`.

Reply:
Oh yeah, nice digging (pun intended) to understand all the work that the data translation step does. I'm okay with your solution then.